src/westpa/core/data_manager.py

```diff
@@ -1119,7 +1119,15 @@ def prepare_segment_restarts(self, segments, basis_states=None, initial_states=N
                 if initial_states is None or basis_states is None:
                     raise ValueError('initial and basis states required for preparing the segments')
                 initial_state = initial_states[segment.initial_state_id]
-                basis_state = basis_states[initial_state.basis_state_id]
+                # Check if it's a start state
```

Comment:
I think we talked about this logic, but could you add a little comment here describing why it's OK to skip adding this data? Or what'll be missing from the per-iteration H5 file? From our discussions, I think the rationale was it's fine to omit this from the h5, because even if this run is completed and started later using the HDF5 framework, the basis states will be used again but the start states will not. So, not storing start states is fine for that reason. If I recall correctly, let's just clearly document that here.

I think you explained to me at one point, but it looks like this is going to leave some segment data in the per-iteration H5 empty. I think this is what you mentioned might cause issues with `westpa.analysis` -- it would be great if you can clearly document why/what specifically those issues might be here. (If this causes problems down the line, or if we decide the above behavior is not acceptable after all, it'd be great to have that written down here so we don't have to rediscover it then)

Reply:
Added in a big block of comments! Thanks!